packages/app/src/breadcrumbs/BreadcrumbsBar.tsx

```diff
@@ -34,13 +34,10 @@ function BreadcrumbsBar(props: Props) {
   } = props;
 
   const { filepath } = useDataContext();
+  const isFullscreen = !!document.fullscreenElement;
 
-  const [isFullscreen, toggleFullScreen] = useToggle(
-    !!document.fullscreenElement
-  );
-  useEventListener(document.documentElement, 'fullscreenchange', () => {
-    toggleFullScreen();
-  });
+  const rerender = useRerender();
+  useEventListener(document, 'fullscreenchange', rerender);
```

Comment:
Simplifying the logic here so that I don't have to add a listener to the element that enters fullscreen (which would require a ref). Note that the `fullscreenchange` event bubbles up to the document.

Reply:
Actually, perhaps a ref would reduce the implicit coupling introduced by the `data-fullscreen-root` field.

You could use the ref to both attach the listener to the relevant fullscreen element and to replace the `querySelector` below.